chia/rpc/wallet_rpc_api.py

```diff
@@ -115,6 +115,7 @@ def get_routes(self) -> Dict[str, Endpoint]:
             "/get_notifications": self.get_notifications,
             "/delete_notifications": self.delete_notifications,
             "/send_notification": self.send_notification,
+            "/sign_message": self.sign_message,
```

Comment:
Is there a need to have 3 separate RPC variants for signing messages? If we want to support data layer singleton signing, will that require a 4th variant?

If we can, I would recommend that we consolidate `nft_sign_message` and `did_sign_message` into this one `sign_message` RPC.

Reply:
I discussed this with @trepca. We decided to build multiple APIs because of the following reasons.
1. We want to have different CLI tools for different types of wallets.
2. Different types of wallets have different ways to get private keys.
The redundant code is just a small part, it's more clear to separate them into different functions.

Reply:
we could break it into two to make managing signing a bit easier on GUI, `sign_message` and `coin_sign_message(type, id, message)`  ?